src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -95,7 +97,42 @@ class PrisonerIepLevelReviewService(
 
   suspend fun handleSyncPostIepReviewRequest(bookingId: Long, syncPostRequest: SyncPostRequest): IepDetail {
     val iepDetail = persistPrisonerIepLevel(bookingId, syncPostRequest)
-    sendEventAndAudit(iepDetail)
+    sendEventAndAudit(
+      iepDetail,
+      eventType = IncentivesDomainEventType.IEP_REVIEW_INSERTED,
+      auditType = AuditType.IEP_REVIEW_ADDED,
+    )
+    return iepDetail
+  }
+
+  suspend fun handleSyncPatchIepReviewRequest(bookingId: Long, id: Long, syncPatchRequest: SyncPatchRequest): IepDetail {
+    if (listOf(syncPatchRequest.iepTime, syncPatchRequest.comment, syncPatchRequest.current).all { it == null }) {
+      throw ValidationException("Please provide fields to update")
+    }
+
+    var prisonerIepLevel: PrisonerIepLevel? = prisonerIepLevelRepository.findById(id)
+    if (prisonerIepLevel == null) {
```

Comment:
You could convert this to `?: run {` but not a big one either way

Reply:
As discussed over Slack I'll leave the `if` as it is.

I've played locally to use `?: run { }` but:
1. seems to cause subsequent lines to complain about the value possibly be `null` (which is weird as I'm throwing an exception and Kotlin should be able to infer that the specific type is a `PrisonerIepLevel` not a `PrisonerIepLevel?` anymore (like it does in the `if` case)
2. it doesn't seem very readable

:)